Add Saas (#89)

gabriel-tessier opened the pull request:
Following the discussion on the issue I created the saas and add chat (chat instead of messenger as Messenger is the band name of the Facebook tool so we will end up with messenger.Messenger!) and social. I only add the example provided in the comment.
Originally I want to do the #134 but there was no saas provider... m(__)m

mingrammer commented:
I thunk the ElasticCloud and ElasticSearch would be better to move under the new `elastic` provider, not saas.

WDYT?

gabriel-tessier commented:
Yes you are right it make more sense to move it to it's own provider. 
If other saas chat and social are ok I'll cancel this PR and make a new one without elastic.

I'll make a new PR for 134.

gabriel-tessier commented:
Got it and thanks for the review.
I'll update and make a PR later today 18:00~19:00 (UTC+9).

mingrammer commented:
Thank you :)

gabriel-tessier commented:
My bad I rebased on master to resolve the conflict it resolve on local but still displayed here.

I can create a new branch and make a different PR if there's problems.

I removed elastic related files and push a new icon for slack.

mingrammer commented on scripts/resource.py:
```diff
@@ -104,6 +104,7 @@ def cleaner_programming(f):
     "alibabacloud": cleaner_alibabacloud,
     "oci": cleaner_oci,
     "programming": cleaner_programming,
+    "saas": cleaner_onprem,
```
Please create a `cleaner_saas`. Redundant code is ok for this.

mingrammer commented:
Slack logo is a little weird.

<img width="276" alt="Screen Shot 2020-05-26 at 11 35 18 AM" src="https://user-images.githubusercontent.com/6178510/82855042-24f7c980-9f45-11ea-991c-873b86b4f437.png">

gabriel-tessier commented:
Thanks for the review, 
- I'll check again the slack logo
- make a cleaner_saas
- and I'll make a new PR from master I feel dirty with the rebase.

mingrammer commented:
Ye, Feel free to make a new PR.

mingrammer commented:
#173 is now merged.
